radicle-httpd/src/api/v1/projects.rs

```diff
@@ -226,17 +224,37 @@ async fn activity_handler(
 /// `GET /projects/:project/tree/:sha/*path`
 async fn tree_handler(
     Extension(ctx): Extension<Context>,
-    Path((project, sha, path)): Path<(Id, Oid, String)>,
+    Path((proj, sha, path)): Path<(Id, Oid, String)>,
 ) -> impl IntoResponse {
     let path = path.strip_prefix('/').ok_or(Error::NotFound)?.to_string();
     let storage = &ctx.profile.storage;
-    let repo = storage.repository(project)?;
-    let tree = radicle_surf::object::tree(&repo.raw().into(), Some(Sha { sha }), Some(path))?;
+    let repo = Repository::open(paths::repository(storage, &proj))?;
+    let tree = repo.tree(sha, &path)?;
+
+    #[derive(Serialize)]
+    struct EntryInfo<'a> {
+        name: &'a str,
+        path: std::path::PathBuf,
+        kind: &'a str,
+        last_commit: &'a radicle_surf::source::commit::Header,
+    }
+
+    let prefix = std::path::Path::new(&path);
+    let entries = tree
+        .entries()
+        .iter()
+        .map(|entry| EntryInfo {
+            name: entry.name(),
+            path: prefix.join(entry.name()),
+            kind: if entry.is_tree() { "tree" } else { "blob" },
+            last_commit: entry.commit(),
+        })
+        .collect::<Vec<EntryInfo>>();
```

Comment:
This is how I think `radicle-httpd` can use `radicle-surf` API without relying on surf's serialization.